[SPARK-13053] [TEST] Unignore tests in InternalAccumulatorSuite

andrewor14 opened the pull request:
These were ignored because they are incorrectly written; they don't actually trigger stage retries, which is what the tests are testing. These tests are now rewritten to induce stage retries through fetch failures.

Note: there were 2 tests before and now there's only 1. What happened? It turns out that the case where we only resubmit a subset of of the original missing partitions is very difficult to simulate in tests without potentially introducing flakiness. This is because the `DAGScheduler` removes all map outputs associated with a given executor when this happens, and we will need multiple executors to trigger this case, and sometimes the scheduler still removes map outputs from all executors.

Commit:
Rewrite test to actually trigger stage retries

This commit actually removes one of the tests, which test stage
retries when only a subset of the original partitions are
submitted. This scenario is difficult to simulate in tests
without introducing flakiness.

Commit:
Fix potential race condition in job completion callback

There's a chance that we don't actually wait until we finish
calling the callback since currently we don't wait for it to
happen. This commit adds a way to do that.

Commit:
Merge branch 'master' of github.com:apache/spark into unignore-accum-test

Commit:
Merge branch 'master' of github.com:apache/spark into unignore-accum-test

andrewor14 commented:
@JoshRosen

SparkQA commented:
**[Test build #50286 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50286/consoleFull)** for PR 10969 at commit [`b0a1f51`](https://github.com/apache/spark/commit/b0a1f511eb550484ce60b3f14755c4163e93feaf).

SparkQA commented:
**[Test build #50286 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50286/consoleFull)** for PR 10969 at commit [`b0a1f51`](https://github.com/apache/spark/commit/b0a1f511eb550484ce60b3f14755c4163e93feaf).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/50286/
Test FAILed.

Commit:
Merge branch 'master' of github.com:apache/spark into unignore-accum-test

andrewor14 commented:
retest this please

SparkQA commented:
**[Test build #50315 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50315/consoleFull)** for PR 10969 at commit [`6322d3a`](https://github.com/apache/spark/commit/6322d3a5693908786f9f73f7306c8cd326416942).

Commit:
Fix flaky AccumulatorSuite#verifyPeakExecutionMemorySet

Events are posted asynchronously so we may not have called
onStageCompleted or onJobEnd etc. before asserting things.
We should wait until all events have been processed before
proceeding.

SparkQA commented:
**[Test build #50317 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50317/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).

SparkQA commented:
**[Test build #50315 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50315/consoleFull)** for PR 10969 at commit [`6322d3a`](https://github.com/apache/spark/commit/6322d3a5693908786f9f73f7306c8cd326416942).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/50315/
Test FAILed.

SparkQA commented:
**[Test build #50317 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50317/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/50317/
Test FAILed.

andrewor14 commented:
retest this please

SparkQA commented:
**[Test build #50328 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50328/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).

SparkQA commented:
**[Test build #50328 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50328/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/50328/
Test FAILed.

andrewor14 commented:
retest this please

SparkQA commented:
**[Test build #50392 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50392/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).

SparkQA commented:
**[Test build #50392 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/50392/consoleFull)** for PR 10969 at commit [`1372766`](https://github.com/apache/spark/commit/13727669abef2f8f4403236b470ca63e1ee92f4a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/50392/
Test PASSed.

andrewor14 commented:
Merging into master.
